collector/lib/CollectionMethod.cpp

```diff
@@ -1,15 +1,26 @@
 #include "CollectionMethod.h"
 
+#include <sstream>
+
+#include <sys/types.h>
+
+#include "Logging.h"
+
 namespace collector {
 
 std::ostream& operator<<(std::ostream& os, CollectionMethod method) {
+  return os << CollectionMethodName(method);
+}
+
+const char* CollectionMethodName(CollectionMethod method) {
   switch (method) {
     case CollectionMethod::EBPF:
-      return os << "ebpf";
+      return "ebpf";
     case CollectionMethod::CORE_BPF:
-      return os << "core_bpf";
+      return "core_bpf";
     default:
-      return os << "unknown(" << static_cast<uint8_t>(method) << ")";
+      CLOG(WARNING) << "Unexpected CollectionMethod: " << static_cast<uint8_t>(method);
+      return "unknown";
```

Comment:
I would argue `ERROR` is a more appropriate severity, since we are not going to be able to continue working with an invalid collection method. I would go as far as to go into `FATAL`, but I'm an extremist.